src/runtime/virtcontainers/container.go

```diff
@@ -1415,8 +1415,6 @@ func (c *Container) cgroupsCreate() (err error) {
 		return fmt.Errorf("Could not create cgroup for %v: %v", c.state.CgroupPath, err)
 	}
 
-	c.config.Resources = resources
```

Comment:
Just so its clear to me, how does the bug fixed int his particular commit manifest?  The container cgroup (if not sandbox cgroups only) is misconfigured on the host (ie, the memory part is being cleared out?)? Or is the final container create request to the agent wrong?

Reply:
Clear the resources info will make the cgroup in the guest cannot get the resources info when the ptr issue is fixed by https://github.com/kata-containers/kata-containers/pull/1759/commits/831224aa224a5d4bf04a21216a4fe8745e2e9549

This is a test with https://github.com/kata-containers/kata-containers/pull/1759/commits/831224aa224a5d4bf04a21216a4fe8745e2e9549 but not https://github.com/kata-containers/kata-containers/pull/1759/commits/0787ea8073ebd311d1758f45be1179f41056aa6c
```
sh -c 'cat > up.json <<-EOF
{
    "metadata": {
        "name": "alpine"
    },
    "log_directory": "/tmp",
    "port_mappings": [{
        "container_port":22,
        "host_port": 8000
    }]
}
EOF'

sh -c 'cat > u.json <<-EOF
{
 "metadata": {
     "name": "ten"
 },
 "image":{
     "image": "docker.io/library/busybox:latest"
 },
 "mounts": [{
    "container_path": "/share",
    "host_path": "/root/tmp"
 }],
 "command": ["top"],
 "linux": {
    "resources": {
        "memory_limit_in_bytes": 1073741824
    }
 }
}
EOF'

crictl pull docker.io/library/busybox:latest

pod=`crictl runp up.json`
c=`crictl  create $pod u.json  up.json`
crictl start $c
crictl exec -t -i $c free -m
              total        used        free      shared  buff/cache   available
Mem:           1978          40        1930           0           7        1917
```

The memory size of container ten is lost because its cgroup limit info is cleared by function cgroupsCreate.

Reply:
Free shows memory available, and isn't restricted to the memory cgroup though, correct?

Can you check the actual cgroup in the guest to verify? I ask, because I was trying to reproduce when running a pod in Kubernetes, and I am seeing the memory cgroup setup as I would expect?

Reply:
> Free shows memory available, and isn't restricted to the memory cgroup though, correct?
> 

The memory size of guest should be default_memory+container size.  https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/sandbox.go#L1811
In this test, the vm size should be 2g(default_memory)+1g(container memory size).
But this vm just have 2g memory.

> Can you check the actual cgroup in the guest to verify? I ask, because I was trying to reproduce when running a pod in Kubernetes, and I am seeing the memory cgroup setup as I would expect? (perhaps because I don't have [831224a](https://github.com/kata-containers/kata-containers/commit/831224aa224a5d4bf04a21216a4fe8745e2e9549) applied?).

The guest cgroup is not affect by this issue because cgroup resources info is send to guest by "process, err := c.sandbox.agent.createContainer(ctx, c.sandbox, c)" is called before "err = c.cgroupsCreate()" is called.

> 
> Similarly, I don't see your original issue; this only manifests if you're doing a container update, is that correct (not an additional container)? I suppose you'd only hit this in kube case if you were doing an in place vertical pod autoscaler?
> 
> I just want to make sure I understand where the errors manifest with this, to assess exposure. Definitely a good catch, and thanks for the details, @teawater !